clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -246,10 +250,15 @@ public void visit(Type field) {
         return hasBuffer.get();
     }
 
-    public static List<ApiKeys> brokerApis() {
-        return Arrays.stream(values())
-            .filter(api -> !api.isControllerOnlyApi)
+    public static EnumSet<ApiKeys> zkBrokerApis() {
+        return apisForListener(ApiMessageType.ListenerType.ZK_BROKER);
+    }
+
+    public static EnumSet<ApiKeys> apisForListener(ApiMessageType.ListenerType listener) {
```

Comment:
given that these `EnumSets` are used so much throughout the code, we should just calculate them once and reuse them.  There are only 3 listener types so it could just be in a static map or something.

Reply:
Makes sense.